src/DotVVM.Framework/Compilation/Styles/StyleRepository.cs

```diff
@@ -53,5 +65,17 @@ public StyleBuilder<HtmlGenericControl> Register(string tagName, Func<StyleMatch
                 return Register<HtmlGenericControl>(m => (string)m.Control.ConstructorParameters[0] == tagName, false);
             }
         }
+
+        private bool isFrozen = false;
+        private void ThrowIfFrozen()
+        {
+            if (isFrozen)
+                throw new InvalidOperationException("The StyleRepository is frozen and can be no longer modified.");
```

Comment:
The exception message is (apart from the type name) the same for all freezable types, can we unify it as a single format string?

Also, the "The" at the beginning seems a little awkward to me, "This" would IMHO be better.

Reply:
Note: `DotvvmConfiguration` itself has a different, older message.

Reply:
> The exception message is (apart from the type name) the same for all freezable types, can we unify it as a single format string?

Yea, we can. But there is the obvious problem where to put that shared function :)

> Note: DotvvmConfiguration itself has a different, older message.

I know, it on purpose - it also explain when does the configuration get frozen. The message for other types is more general as they can be also use in different context.